File: medium/60.第k个排列/60.第k个排列.js
/*
 * @lc app=leetcode.cn id=60 lang=javascript
 *
 * [60] 第k个排列
 */
/**
 * @param {number} n
 * @param {number} k
 * @return {string}
 */
var getPermutation = function(n, k) {
  const arr = Array.from({ length: n }, (_, i) => i + 1)
  const res = []
  let pos = k - 1
  let fac = arr.reduce((a, b) => a * b)
  for (let j = n; j >= 1; --j) {
    fac /= j
    res.push(arr.splice(~~(pos / fac), 1)[0])
    pos %= fac
  }
  return res.join('')
}

File: medium/60.第k个排列/60.第k个排列.py
#
# @lc app=leetcode.cn id=60 lang=python3
#
# [60] 第k个排列
#
class Solution:
    def getPermutation(self, n: int, k: int) -> str:
        arr = [x for x in range(1, n + 1)]
        res = []
        pos = k - 1
        fac = 1
        for n in arr:
            fac *= n
        for j in range(n, 0, -1):
            fac //= j
            res.append(str(arr.pop(pos // fac)))
            pos %= fac
        return ''.join(res)
